CppLinq/cpplinq.hpp

```diff
@@ -3305,10 +3560,76 @@ namespace cpplinq
             typedef             std::vector<std::pair<key_type, size_type>>     keys_type           ;
             typedef             std::vector<value_type>                         values_type         ;
 
+            typedef typename    keys_type::const_iterator                       keys_iterator_type  ;
             typedef typename    values_type::const_iterator                     values_iterator_type;
 
-            template<typename TRange, typename TSelector>
-            CPPLINQ_METHOD lookup (size_type capacity, TRange range, TSelector selector)
+            template<typename TRange, typename TSelector>
+            CPPLINQ_METHOD lookup (size_type capacity, TRange range, TSelector selector)
```

Comment:
Merged most of the functionality in the overloads into one method in a later commit.